### PR TITLE
bayesian-analysis-toolkit v1.0.0 (new formula)

### PR DESCRIPTION
**Pull request by gipert**
As a `bat` formula already exists, I chose `bat-stat`, better ideas are very welcome...

Parallelism in BAT is currently implemented with OpenMP, which, as we know, is not part of Clang yet. So I left that option for Linux systems, as, I think, no one would want to install GCC as a dependency on macOS (also because BAT is just a set of libraries, the user would have to compile his project with the same GCC...).

### CODE REVIEW

**Comment by gipert** [comment]
Uhmm the only error I get from `brew audit bat-stat --online --new-formula` on my laptop is
```
gipert/physics/bat-stat:
  * Non-libraries were installed to "/usr/local/opt/bat-stat/lib"
    Installing non-libraries to "lib" is discouraged.
    The offending files are:
      /usr/local/opt/bat-stat/lib/libBATmodels.rootmap
            /usr/local/opt/bat-stat/lib/libBATmtf.rootmap
            /usr/local/opt/bat-stat/lib/libBATmtf_rdict.pcm
            /usr/local/opt/bat-stat/lib/libBATmodels_rdict.pcm
            /usr/local/opt/bat-stat/lib/libBAT.rootmap
            /usr/local/opt/bat-stat/lib/libBAT_rdict.pcm
Error: 1 problem in 1 formula
```
Those are [ROOT](https://root.cern.ch) dictionary files, the official recommendation is to put them along with the libraries, because ROOT then, during an interactive session, looks for them in `LD_LIBRARY_PATH`.

And oh, yes, I already added OS checks, but I guess I can remove them and let the Linuxbrew maintainers add them in their fork, let me know.

**Comment by gipert** [comment]
Hope that someone will have time to review this PR, it would be a pity to let it die...

**Comment by fxcoudert** [comment]
@BrewTestBot test this please

File: Formula/bat-stat.rb [hunk 1]
```diff
@@ -0,0 +1,36 @@
+class BatStat < Formula
+  desc "Bayesian analysis toolkit"
+  homepage "https://bat.mpp.mpg.de/"
+  url "https://github.com/bat/bat/releases/download/v1.0.0/BAT-1.0.0.tar.gz"
+  sha256 "620e8069d85f18f8504137823621cbb578fa5b159e3074f0f2379ac295dd1482"
+
+  depends_on "root"
+  depends_on "cuba" => :recommended
```
Comment (fxcoudert) [inline Formula/bat-stat.rb]: Make cuba required, not recommended.

File: Formula/bat-stat.rb [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+
+  option "with-multithreading", "Enable OpenMP multithreading support (gcc only)" if OS.linux?
```
Comment (fxcoudert) [inline Formula/bat-stat.rb]: No option, no linux code.

File: Formula/bat-stat.rb [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+      opts << "--with-cuba"
+    end
+
+    if build.with?("multithreading") && OS.mac?
+      odie "Multithreading support is not available on OSX for this formula"
+    end
+
+    opts << "--enable-parallel" if build.with? "multithreading"
+
+    system "./configure", *opts
```
Comment (fxcoudert) [inline Formula/bat-stat.rb]: Inline all arguments to configure. No addition `opts` variable.

File: Formula/bat-stat.rb [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+    system "make", "install"
+    (share/"BAT/examples").install "examples/basic"
```
Comment (fxcoudert) [inline Formula/bat-stat.rb]: `pkgshare.install`

File: Formula/bat-stat.rb [hunk 5]
```diff
@@ -0,0 +1,36 @@
...
+    (share/"BAT/examples").install "examples/advanced"
+  end
+
+  test do
+    system "#{share}/BAT/examples/basic/binomial/runEfficiency"
```
Comment (fxcoudert) [inline Formula/bat-stat.rb]: What is this test doing? Can we check its output?

**Review by fxcoudert — CHANGES_REQUESTED** [review]
How about `bayesian-analysis-toolkit` as formula name?

**Comment by DomT4** [comment]
> I'd welcome your opinion on ROOT files located in lib here…

Fairly grim. I wonder if it's possible to convince `root` to look elsewhere by default. I could live with `share/root/dictionaries` or something similar. Looking the other way on putting these files in `lib` naked is going to take some convincing, given how few exceptions we make.

File: Formula/bat-stat.rb (hunk 5 above)
Comment (gipert) [reply Formula/bat-stat.rb]: The official test suite can be run with a `make check` in the source build tree, as described [here](https://github.com/bat/bat/tree/master/test). I don't know how to do that in the `test do ... end` section (I don't know if it's even possible), so I use one of the examples (there are many of them) as a test. The `runEfficiency` one just represents one possible (very simple) statistical analysis that can be implemented with BAT. [[docs]](https://github.com/bat/bat/tree/master/examples/basic/binomial)

What do you think?

**Comment by gipert** [comment]
> How about bayesian-analysis-toolkit as formula name?

Ok for me, I thought it was too long

> I wonder if it's possible to convince root to look elsewhere by default. I could live with share/root/dictionaries or something similar. Looking the other way on putting these files in lib naked is going to take some convincing, given how few exceptions we make.

I noticed that `root` puts its own libraries and dictionaries in `lib/root` instead of `lib`, and that's why the `brew audit` error does not show up, I guess.

**Comment by DomT4** [comment]
> I noticed that root puts its own libraries and dictionaries in lib/root instead of lib

If we manually move the dictionaries to the same path inside this keg does `root` find them successfully?

**Comment by gipert** [comment]
Quoting from the ROOT manual:

>This tool [...] generates in output the dictionary C++ code and a `pcm` file. This latter file is fundamental for the correct functioning of the dictionary at runtime. It should be located in the directory where the shared library is installed in which the compiled dictionary resides.

I didn't find any info about putting libraries and dictionaries under different paths so far, so the only thing I can do is to move both libraries and dictionaries inside, e.g., `lib/bat`. Would it be OK?

**Comment by fxcoudert** [comment]
@gipert that would work, I think

**Comment by gipert** [comment]
What about now? Being more rigorous, `lib/bayesian-analysis-toolkit` would be the right path, but I feel it's way too long...

**Pull request closed without merging**